diff --git a/kernel/selinux/selinux.h b/kernel/selinux/selinux.h
index 750f2c26..46bf9527 100644
--- a/kernel/selinux/selinux.h
+++ b/kernel/selinux/selinux.h
@@ -5,6 +5,13 @@
 #include "linux/version.h"
 #include "linux/cred.h"
 
+// TODO: rename to "ksu"
+#define KERNEL_SU_DOMAIN "su"
+#define KERNEL_SU_FILE "ksu_file"
+
+#define KERNEL_SU_CONTEXT "u:r:" KERNEL_SU_DOMAIN ":s0"
+#define KSU_FILE_CONTEXT "u:object_r:" KERNEL_SU_FILE ":s0"
+
 void setup_selinux(const char *);
 
 void setenforce(bool);
